app/controllers/feedbacks_controller.rb

```diff
@@ -0,0 +1,74 @@
+class FeedbacksController < ApplicationController
+  before_action :set_feedback, only: [:show, :edit, :update, :destroy]
+  before_action :set_feedbackable, only: [:new, :create]
+
+  def index
+    authorize Feedback
+    @feedbacks = policy_scope(Feedback)
+    if params[:group_offer_id]
+      @feedbacks = @feedbacks.where(feedbackable_id: params[:group_offer_id])
+      @volunteer = current_user.volunteer || @feedbacks.first.volunteer
+    elsif params[:assignment_id]
+      @feedbacks = @feedbacks.where(feedbackable_id: params[:assignment_id])
+      @volunteer = Assignment.find(params[:assignment_id]).volunteer
+    elsif params[:volunteer_id]
+      @feedbacks = @feedbacks.where(volunteer_id: params[:volunteer_id])
+      @volunteer = Volunteer.find(params[:volunteer_id])
+    end
```

Comment:
Hmm it looks like this could be simplified to `params[:feedbackable]` and something like `@feedbackable = @feedbacks.first.feedbackable`

Not sure how important the `current_user.volunteer` case is though.

Reply:
Are the params here used for filtering in the UI or something?

Reply:
there is three different nested routes

```ruby
  resources :volunteers do
    resources :feedbacks
  end
  resources :assignments do
    resources :feedbacks
  end
  resources :group_offers do
    resources :feedbacks
  end
```